core/shared/src/main/scala/scalaz/zio/Duration.scala

```diff
@@ -0,0 +1,25 @@
+package scalaz.zio
+
+sealed trait Duration {
+  def toDuration: scala.concurrent.duration.Duration
```

Comment:
I don't like this. I think we should not support converting into Scala type. Instead we should have the subset of useful and safe methods on `Finite`.

Reply:
@jdegoes @jczuchnowski this seems like a safe subset
```scala
  def isFinite(): Boolean
  def gt(other: Duration)
  def gteq(other: Duration)
  def lt(other: Duration)
  def lteq(other: Duration)
  def min(other: Duration): Duration
  def max(other: Duration): Duration
```

Other things throw for `Inf` or don't make much sense for (potentially) `Inf`, like `+` or `*`. What do you think?

Reply:
I'd delete `isFinite()`. Also if some of these are subsumed by `Ordered[Duration]`, we can omit them.